scout/adapter/mongo/variant_loader.py

```diff
@@ -658,85 +666,90 @@ def load_variants(
-                vep_header=vep_header,
-                category=category,
-                sample_info=sample_info,
-                custom_images=custom_images,
-                local_archive_info=local_archive_info,
-            )
-        except Exception as error:
-            LOG.exception("unexpected error")
-            LOG.warning("Deleting inserted variants")
-            self.delete_variants(case_obj["_id"], variant_type)
-            raise error
+            # Parse the necessary headers from vcf file
+            rank_results_header = parse_rank_results_header(vcf_obj)
+
+            local_archive_info = parse_local_archive_header(vcf_obj)
+
+            vep_header = parse_vep_header(vcf_obj)
+            if vep_header:
+                LOG.debug("Found VEP header %s", "|".join(vep_header))
+
+            # This is a dictionary to tell where ind are in vcf
+            individual_positions = {ind: i for i, ind in enumerate(vcf_obj.samples)}
+
+            # Dictionary for cancer analysis
+            sample_info = {}
+            if category in ("cancer", "cancer_sv"):
+                for ind in case_obj["individuals"]:
+                    if ind["phenotype"] == 2:
+                        sample_info[ind["individual_id"]] = "case"
+                    else:
+                        sample_info[ind["individual_id"]] = "control"
+
+            # Check if a region should be uploaded
+            region = ""
+            if gene_obj:
+                chrom = gene_obj["chromosome"]
+                # Add same padding as VEP
+                start = max(gene_obj["start"] - 5000, 0)
+                end = gene_obj["end"] + 5000
+            if chrom:
+                # We want to load all variants in the region regardless of rank score
+                rank_threshold = rank_threshold or -1000
+                if not (start and end):
+                    raise SyntaxError("Specify chrom start and end")
+                region = "{0}:{1}-{2}".format(chrom, start, end)
+            else:
+                rank_threshold = rank_threshold or 0
+
+            variants = vcf_obj(region)
 
-        self.update_variant_rank(case_obj, variant_type, category=category)
+            try:
+                nr_inserted = self._load_variants(
+                    variants=variants,
+                    variant_type=variant_type,
+                    case_obj=case_obj,
+                    individual_positions=individual_positions,
+                    rank_threshold=rank_threshold,
+                    institute_id=institute_id,
+                    build=build,
+                    rank_results_header=rank_results_header,
+                    vep_header=vep_header,
+                    category=category,
+                    sample_info=sample_info,
+                    custom_images=custom_images,
+                    local_archive_info=local_archive_info,
+                    gene_to_panels=gene_to_panels,
```

Comment:
The last three args are here since we can now reuse these for both files. They could actually be made even earlier in `load_case()` but that will be even more passing stuff around (and changes to where and when `build` is defaulted) for relatively little processing.